src/main/java/lol/ui/BattlefieldView.java

```diff
@@ -36,6 +36,12 @@ public BattlefieldView(Battlefield battlefield, Stage stage) {
     sprites = new Sprites();
   }
 
+  public void generateNewBattlefieldView(Battlefield battlefield) {
```

Comment:
Instead of having this method, wouldn't it be more convenient to just recreate the object BattlefieldView each time we start a new game? That might solve your first issue as well.